### PR TITLE
Extents/Offsets move definitions from header to cpp

### PR DESCRIPTION
**Pull request by lukasm91**
## Technical Description

Final task for extents/offsets is moving definitions from header to cpp to reduce dependencies when changing one of the definitions. Basically all files depend on `Offsets.h`.

No code was changed in the implementations.

### Dependencies

no dependencies

### CODE REVIEW

**Comment by lukasm91** [comment]
launch jenkins

File: dawn/src/dawn/IIR/Extents.h [hunk 1]
```diff
@@ -41,45 +35,34 @@ class Extent {
...
-  bool isPointwise() const { return plus_ == 0 && minus_ == 0; }
+  bool isPointwise() const;
   /// @}
   //
+
 private:
   int minus_;
   int plus_;
-};
+}; // namespace iir
```
Comment (lukasm91) [inline dawn/src/dawn/IIR/Extents.h]:
```suggestion
};
```

File: dawn/src/dawn/AST/Offsets.h [hunk 2]
```diff
@@ -195,39 +159,28 @@ auto offset_dispatch(HorizontalOffset const& hOffset, CartFn const& cartFn,
...
+  bool operator!=(Offsets const& other) const;
+  Offsets& operator+=(Offsets const& other);
+
+  bool isZero() const;
 
 private:
   HorizontalOffset horizontalOffset_;
   int verticalOffset_ = 0;
-};
+}; // namespace dawn::ast
```
Comment (lukasm91) [inline dawn/src/dawn/AST/Offsets.h]:
```suggestion
};
```

**Comment by havogt** [comment]
launch jenkins

**Comment by lukasm91** [comment]
launch jenkins

**Pull request merged**